controllers/ovndbcluster_controller.go

```diff
@@ -579,8 +571,122 @@ func (r *OVNDBClusterReconciler) reconcileServices(
+			}
+
+		}
+	}
+
+	podList, err = ovndbcluster.OVNDBPods(ctx, instance, helper, serviceLabels)
+	if err != nil {
+		return ctrl.Result{}, err
+	}
+	// Assuming that the replicas is greater than 0 AND networkAttachment != ""
+	// dbAddress will contain ovsdbserver-(nb|sb).openstack.svc
+	// if replicas == 0 or networkAttachment == "" dbAddress will be empty.
+	// Meaning that Status.DBAddress is not ready.
+	dbAddress := ""
+
+	// DNSData is needed to provide DNS outside the cluster (edpm nodes) which needs to have
+	// spec.NetworkAttachment != ""
+	if instance.Spec.NetworkAttachment != "" {
+		for i, ovnPod := range podList.Items {
+			if i >= int(*(instance.Spec.Replicas)) {
+				break
+			}
+			var dnsName string
+			var dnsIP string
+			var hostnames []string
+			dnsName = v1beta1.DNSPrefix + "-" + ovnPod.Name
+			// Get Hostname
+			svc, err := service.GetServiceWithName(
+				ctx,
+				helper,
+				ovnPod.Name,
+				ovnPod.Namespace,
+			)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
+			hostname := svc.ObjectMeta.Annotations[infranetworkv1.AnnotationHostnameKey]
+			hostnames = append(hostnames, hostname)
+
+			// Get IP
+			net_stat, err := nad.GetNetworkStatusFromAnnotation(ovnPod.Annotations)
+			if err != nil {
+				Log.Info(fmt.Sprintf("Error while getting the NAD for pod %s: %v", ovnPod.Name, err))
+				return ctrl.Result{}, err
+			}
+			for _, v := range net_stat {
+				if strings.Contains(v.Name, instance.Spec.NetworkAttachment) {
+					// Return only IPv4 at the moment
+					if !strings.Contains(v.IPs[0], ":") {
+						dnsIP = v.IPs[0]
+						break
+					}
+				}
+			}
+
+			if len(dnsIP) == 0 {
+				Log.Info(fmt.Sprintf("Error while getting pod %s %s IP, IP is empty", ovnPod.Name, instance.Spec.NetworkAttachment))
+				continue
+			}
+
+			// Create DNSRecord
+			var DNSHosts []infranetworkv1.DNSHost
+			// ovsdbserver-(nb|sb)-n entry
+			DNSHost := infranetworkv1.DNSHost{}
+			DNSHost.IP = dnsIP
+			DNSHost.Hostnames = hostnames
+			// ovsdbserver-(sb|nb) entry
+			headlessDnsHostname := ovndbcluster.ServiceNameSB + "." + instance.Namespace + ".svc"
+			if instance.Spec.DBType == v1beta1.NBDBType {
+				headlessDnsHostname = ovndbcluster.ServiceNameNB + "." + instance.Namespace + ".svc"
+			}
+			dbAddress = fmt.Sprintf("tcp:%s:%d", headlessDnsHostname, svc.Spec.Ports[0].Port)
+			DNSHostCname := infranetworkv1.DNSHost{}
+			DNSHostCname.IP = dnsIP
+			DNSHostCname.Hostnames = append(DNSHostCname.Hostnames, headlessDnsHostname)
+			DNSHosts = append(DNSHosts, DNSHost)
+			DNSHosts = append(DNSHosts, DNSHostCname)
+
+			// Create DNSData
+			DNSData := &infranetworkv1.DNSData{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      dnsName,
+					Namespace: ovnPod.Namespace,
+					Labels:    serviceLabels,
+				},
+			}
+			_, err = controllerutil.CreateOrPatch(ctx, helper.GetClient(), DNSData, func() error {
+				DNSData.Spec.Hosts = DNSHosts
+				DNSData.Spec.DNSDataLabelSelectorValue = "dnsdata"
```

Comment:
would be better to use DNSDataLabelSelectorValue from DNSMasq instance instead of hardcoding here, could be done in follow ups too but good to keep a NOTE for it here.

Reply:
Will add a note to add a follow up, thanks